diff -ruN build_orig/lua-5.1.4/src/luaconf.h build/lua-5.1.4/src/luaconf.h
--- build_orig/lua-5.1.4/src/luaconf.h	2008-02-11 16:25:08.000000000 +0000
+++ build/lua-5.1.4/src/luaconf.h	2011-02-07 04:19:57.000000000 +0000
@@ -501,8 +501,8 @@
 ** ===================================================================
 */
 
-#define LUA_NUMBER_DOUBLE
-#define LUA_NUMBER	double
+#define LUA_NUMBER_FLOAT
+#define LUA_NUMBER	float
 
 /*
 @@ LUAI_UACNUMBER is the result of an 'usual argument conversion'
@@ -518,7 +518,7 @@
 @@ LUAI_MAXNUMBER2STR is maximum size of previous conversion.
 @@ lua_str2number converts a string to a number.
 */
-#define LUA_NUMBER_SCAN		"%lf"
+#define LUA_NUMBER_SCAN		"%f"
 #define LUA_NUMBER_FMT		"%.14g"
 #define lua_number2str(s,n)	sprintf((s), LUA_NUMBER_FMT, (n))
 #define LUAI_MAXNUMBER2STR	32 /* 16 digits, sign, point, and \0 */
diff -ruN build_orig/luadec/print.c build/luadec/print.c
--- build_orig/luadec/print.c	2009-03-16 13:53:48.000000000 +0000
+++ build/luadec/print.c	2011-02-07 04:19:57.000000000 +0000
@@ -42,7 +42,7 @@
  */
 
 char* getupval(Function * F, int r) {
-	if (F->f->upvalues) {
+  if (F->f->upvalues && r < F->f->sizeupvalues) {
 		return (char*)getstr(F->f->upvalues[r]);
 	} else {
 		char* s = malloc(20);
diff -ruN build_orig/luadec/proto.c build/luadec/proto.c
--- build_orig/luadec/proto.c	2009-02-07 13:04:34.000000000 +0000
+++ build/luadec/proto.c	2011-02-07 04:20:25.000000000 +0000
@@ -1,6 +1,3 @@
-
-#include "proto.h"
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
